gulpfile.js

```diff
@@ -6,245 +6,131 @@
 const glob = require('glob');
 const execSync = require('child_process').execSync;
 const jsonStableStringify = require('json-stable-stringify');
 
-var mappings = require('./codegen_mappings.json');
-
+const azureSDKForNodeRepoRoot = __dirname;
 const defaultAutoRestVersion = '1.2.2';
 var usingAutoRestVersion;
-const specRoot = args['spec-root'] || 'https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification';
-const project = args['project'];
+let azureRestAPISpecsRoot = args['azure-rest-api-specs-root'] || path.resolve(azureSDKForNodeRepoRoot, '..', 'azure-rest-api-specs');
+const package = args['package'];
 const use = args['use'];
-var modeler = 'Swagger';
 const regexForExcludedServices = /\/(intune|documentdbManagement|insightsManagement|insights|search)\//i;
 
-function getAutorestVersion(version) {
-  if (!version) version = 'latest';
-  let getVersion, execHelp;
-  let result = true;
-  try {
-    let getVersionCmd = `autorest `;
-    let execHelpCmd = `autorest --help`;
-    console.log(getVersionCmd);
-    getVersion = execSync(getVersionCmd, { encoding: 'utf8' });
-    //console.debug(getVersion);
-    console.log(execHelpCmd);
-    execHelp = execSync(execHelpCmd, { encoding: 'utf8' });
-    //console.debug(execHelp);
-  } catch (err) {
-    result = false;
-    console.log(`An error occurred while getting the "${version}" of autorest and executing "autorest --help":\n ${util.inspect(err, { depth: null })}.`);
-  }
-  return result;
-}
+function findReadmeNodejsMdFilePaths(azureRestAPISpecsRoot) {
+  const nodejsReadmeFilePaths = [];
+
+  // Find all of the readme.nodejs.md files within the azure-rest-api-specs/specification folder.
+  const specificationFolderPath = path.resolve(azureRestAPISpecsRoot, 'specification');
 
-function deleteFolderRecursive(path) {
-  if (fs.existsSync(path)) {
-    fs.readdirSync(path).forEach(function (file, index) {
-      var curPath = path + '/' + file;
-      if (fs.lstatSync(curPath).isDirectory()) { // recurse
-        deleteFolderRecursive(curPath);
-      } else { // delete file
-        fs.unlinkSync(curPath);
+  const folderPathsToSearch = [specificationFolderPath];
+  while (folderPathsToSearch.length > 0) {
+    const folderPathToSearch = folderPathsToSearch.pop();
+
+    const folderEntryPaths = fs.readdirSync(folderPathToSearch);
+    for (let i = 0; i < folderEntryPaths.length; ++i) {
+      const folderEntryPath = path.resolve(folderPathToSearch, folderEntryPaths[i]);
+      const folderEntryStats = fs.lstatSync(folderEntryPath);
+      if (folderEntryStats.isDirectory()) {
+        folderPathsToSearch.push(folderEntryPath);
+      }
+      else if (folderEntryStats.isFile()) {
+        const fileName = path.basename(folderEntryPath);
+        if (fileName === 'readme.nodejs.md') {
+          nodejsReadmeFilePaths.push(folderEntryPath);
+        }
       }
-    });
-    fs.rmdirSync(path);
+    }
   }
-}
 
-function clearProjectBeforeGenerating(projectDir) {
```

Comment:
W.r.t cleaning projects before generating there are sdks like azure-batch, azure-keyvault and datalake analytics/filesystem that have some hand written files along with the autogenerated files. We should make sure that Autorest does not accidentally delete them.

Reply:
I'll verify this. Hold on.

Reply:
Yup, I checked and the handwritten stuff stayed put.